certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -67,8 +67,30 @@ def __eq__(self, other):
         return False
 
     def set_parameters(self, parameters):
-        """Sets the parameters for DirectiveNode"""
-        self.parameters = parameters
+        """
+        Sets parameters of a DirectiveNode or BlockNode object.
+
+        :param list parameters: List of all parameters for the node to set.
+        """
+        orig_params = self._aug_get_params(self.metadata["augeaspath"])
+
+        # Clear out old parameters
+        for _ in orig_params:
+            # When the first parameter is removed, the indices get updated
+            param_path = "{}/arg[1]".format(self.metadata["augeaspath"])
+            self.parser.aug.remove(param_path)
+        # Insert new ones
+        for pi, _ in enumerate(parameters):
+            param_path = "{}/arg[{}]".format(self.metadata["augeaspath"], pi+1)
+            self.parser.aug.set(param_path, parameters[pi])
```

Comment:
Super new to augeas syntax... parameters are set at a particular index? Surprised that we have to remove the old parameters before new ones can be added.

Reply:
Yeah, the parameters are set at a particular index. We don't strictly have to remove and re-add the parameters in order to change them, but if `len(new_parameters) < len(old_parameters)` we have to explicitly clear the postponing, previously set indices. I think this approach helps to keep the code more readable, and less error prone.